### PR TITLE
[optimization](array-type) support the decimal/datetime as the nest t…

### PR DESCRIPTION
**Pull request by carlvinhust2012**
# Proposed changes
1. This pr is used to support the decimal/datetime as the nest type of array in print_value.
2. you can use 'select into outfile' to export the decimal/datetime nest type.
Issue Number: close #7570

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
3. Has unit tests been added:
    - [ ] Yes
    - [x] No
    - [ ] No Need
4. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
5. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
6. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: regression-test/suites/export/test_array_export.groovy [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        """
+        File[] files = path.listFiles()
+        assert files.length == 1
+        List<String> outLines = Files.readAllLines(Paths.get(files[0].getAbsolutePath()), StandardCharsets.UTF_8);
+        List<String> baseLines = Files.readAllLines(Paths.get("""${context.config.dataPath}/export/test_array_export.out"""), StandardCharsets.UTF_8)
+        for (int j = 0; j < outLines.size(); j ++) {
+            String[] outLine = outLines.get(j).split("\t")
+            String[] baseLine = baseLines.get(j + 2).split("\t")
+            for (int slotId = 0; slotId < outLine.size(); slotId ++) {
+                assert outLine[slotId] == baseLine[slotId]
```
Comment (spaces-X) [inline regression-test/suites/export/test_array_export.groovy:119]: P0 regression failed here caused by format and precision issues when reading files

Comment (carlvinhust2012) [reply regression-test/suites/export/test_array_export.groovy:119]: ok，I will re-check.

**Review by spaces-X — APPROVED** [review]
LGTM

**Review by xy720 — APPROVED** [review]
LGTM

**Pull request merged**